client/sites-dashboard/components/sites-dashboard.tsx

```diff
@@ -134,21 +134,26 @@ export function SitesDashboard( { queryParams: { search, status } }: SitesDashbo
 	);
 }
 
-function handleTabSelect( tabName: string ) {
-	page(
-		'all' === tabName
-			? removeQueryArgs( window.location.pathname + window.location.search, 'status' )
-			: addQueryArgs( window.location.pathname + window.location.search, { status: tabName } )
-	);
-}
-
-function handleSearch( rawTerm: string ) {
-	const trimmedTerm = rawTerm.trim();
-	if ( trimmedTerm.length ) {
-		page(
-			addQueryArgs( window.location.pathname + window.location.search, { search: trimmedTerm } )
-		);
+/**
+ * Updates a query param used by the sites dashboard, causing a page navigation.
+ * Param will be removed if it is empty or matches its default value.
+ *
+ * @param paramName name of the param being updated
+ * @param paramValue new value for the param
+ * @param defaultValue if the new param matches the default it'll be removed
+ */
+function handleQueryParamChange(
+	paramName: keyof SitesDashboardQueryParams,
+	paramValue: string | null,
+	defaultValue = ''
```

Comment:
I don't think we should have default values because we'll either upsert a query parameter or delete it. None are actually required, so I wonder what the motivation for this additional argument is.

Reply:
> I don't think we should have default values because we'll either upsert a query parameter or delete it. None are actually required, so I wonder what the motivation for this additional argument is.

Removed with cb001749e323624c07447087a6ac9904884207b0

Reply:
My original version of this refactor looked a lot like what @danielbachhuber implemented in https://github.com/Automattic/wp-calypso/commit/250537e2f505e988ae1ae39cdbeb904da1e04b01 and https://github.com/Automattic/wp-calypso/commit/cb001749e323624c07447087a6ac9904884207b0. In retrospect I'm glad the "magic" has been removed from `handleQueryParamChange`.